### PR TITLE
Support --enable-proposed-api through argv.json

### LINKED ISSUE #99775: Support --enabled-proposed-api thorugh argv.json
Currently, extensions which use proposed API's cannot be run on product builds without passing the `--enabled-proposed-api [...ext-id's...]` CLI flag. This makes long-term dogfooding of proposed API's and the extensions which use them very inconvenient for people (such as myself) who tend to open VS Code without the CLI and thus cannot pass CLI arguments. 

For instance, I have made extensions using the timeline view that I'd like to use in daily life to get more testing in to the Timeline API and provide feedback, but it's not worth it to manually edit `product.json` every day. I think as a "big kid" VS Code user I should be able to opt-in to enabling unstable features that might break at any point (that's what Insiders is all about, right?).

To this end, it would be nice if we could additionally enable proposed API's in the existing `argv.json` file. The main concern with doing this is that bad actors could in theory edit this file from the extension, but they could already edit `product.json`, so doing this isn't really adding any new holes. 

cc @eamodio @jrieken

### PR DESCRIPTION
**Pull request by JacksonKearl**
Closes #99775

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -201,30 +201,43 @@ function configureCommandlineSwitchesSync(cliArgs) {
 		SUPPORTED_ELECTRON_SWITCHES.push('force-renderer-accessibility');
 	}
 
+	const SUPPORTED_MAIN_PROCESS_SWITCHES = [
+		// Persistantly enable proposed api via argv.json: https://github.com/microsoft/vscode/issues/99775
```
Comment (gjsjohnmurray) [inline src/main.js:205]: Persistantly -> Persistently

File: src/vs/workbench/electron-sandbox/desktop.contribution.ts [hunk 2]
```diff
@@ -343,6 +343,13 @@ import { IJSONSchema } from 'vs/base/common/jsonSchema';
 			'force-color-profile': {
 				type: 'string',
 				markdownDescription: nls.localize('argv.forceColorProfile', 'Allows to override the color profile to use. If you experience colors appear badly, try to set this to `srgb` and restart.')
+			},
+			'enable-proposed-api': {
+				type: 'array',
+				description: nls.localize('argv.enebleProposedApi', "Enable proposed APIs for a list of extension id's (such as \`vscode.git\`). Proposed APIs are unstable and subject to breaking without warning at any time. This should only be set for extension development and testing purposes."),
```
Comment (gjsjohnmurray) [inline src/vs/workbench/electron-sandbox/desktop.contribution.ts:349]: id's -> ids

Comment (JacksonKearl) [reply src/vs/workbench/electron-sandbox/desktop.contribution.ts]: Thanks!

File: src/main.js (hunk 1 above)
Comment (JacksonKearl) [reply src/main.js]: Thanks!

File: src/main.js [hunk 3]
```diff
@@ -201,30 +201,43 @@ function configureCommandlineSwitchesSync(cliArgs) {
 		SUPPORTED_ELECTRON_SWITCHES.push('force-renderer-accessibility');
 	}
 
+	const SUPPORTED_MAIN_PROCESS_SWITCHES = [
+		// Persistently enable proposed api via argv.json: https://github.com/microsoft/vscode/issues/99775
+		'enable-proposed-api'
+	];
+
```
Comment (bpasero) [inline src/main.js:208]:
```suggestion
	const SUPPORTED_MAIN_PROCESS_SWITCHES = [

		// Persistently enable proposed api via argv.json: https://github.com/microsoft/vscode/issues/99775
		'enable-proposed-api'
	];

```

File: src/main.js [hunk 4]
```diff
@@ -201,30 +201,43 @@ function configureCommandlineSwitchesSync(cliArgs) {
...
+			// Others
+			else if (argvValue === true || argvValue === 'true') {
+				if (argvKey === 'disable-hardware-acceleration') {
+					app.disableHardwareAcceleration(); // needs to be called explicitly
+				} else {
+					app.commandLine.appendSwitch(argvKey);
+				}
+			}
+		}
+		// Append main process flags to process.argv
```
Comment (bpasero) [inline src/main.js:233]:
```suggestion
		}

		// Append main process flags to process.argv
```

**Review by bpasero — APPROVED** [review]

**Pull request merged**